modules/monitoring-setting-up-pod-topology-spread-constraints.adoc

```diff
@@ -0,0 +1,189 @@
+
+. Edit the `cluster-monitoring-config` config map in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Add the following settings under the `data/config.yaml` field to configure pod topology spread constraints:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    <component>: # <1>
+      topologySpreadConstraints:
+      - maxSkew: <n> # <2>
+        topologyKey: monitoring # <3>
+        whenUnsatisfiable: <value> # <4>
+        labelSelector:
+          matchLabels: # <5>
+            <label>
+----
+<1> Specify a name of the component for which you want to set up pod topology spread constraints.
+<2> Specify a numeric value for `maxSkew`, which defines the degree to which pods are allowed to be unevenly distributed.
+This field is required, and the value must be greater than zero.
+The value specified has a different effect depending on what value you specify for `whenUnsatisfiable`.
+<3> Specify a key of node labels for `topologyKey`.
+This field is required.
+Nodes that have a label with this key and identical values are considered to be in the same topology.
+The scheduler tries to put a balanced number of pods into each domain.
+<4> Specify a value for `whenUnsatisfiable`.
+This field is required.
+Available options are `DoNotSchedule` and `ScheduleAnyway`.
+Specify `DoNotSchedule` if you want the `maxSkew` value to define the maximum difference allowed between the number of matching pods in the target topology and the global minimum.
+Specify `ScheduleAnyway` if you want the scheduler to still schedule the pod but to give higher priority to nodes that might reduce the skew.
+<5> Specify a label for `matchLabels` to identify the set of matching pods to which to apply the constraints.
++
+.Example configuration for Prometheus
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      topologySpreadConstraints:
+      - maxSkew: 1
+        topologyKey: monitoring
+        whenUnsatisfiable: DoNotSchedule
+        labelSelector:
+          matchLabels:
+            app.kubernetes.io/name: prometheus
+----
+
+. Save the file to apply the changes automatically.
++
+[WARNING]
+====
+When you save changes to the `cluster-monitoring-config` config map, the pods and other resources in the `openshift-monitoring` project might be redeployed.
+The running monitoring processes in that project might also restart.
+====
+
+* *To configure pod topology spread constraints for user-defined monitoring:*
+endif::openshift-dedicated,openshift-rosa[]
+
+. Edit the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` project:
```

Comment:
This snippet appears to be duplicated because the configuration is identical in the user-defined project. It may suffice to explain only the differences in the configuration map:

```
Edit the `cluster-monitoring-config` config map in the `openshift-monitoring` project:
and 
Edit the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` project:
```
Because the example is the same, just change the name and namespace:
```
data:
apiVersion: v1
kind: ConfigMap
metadata:
  name: user-workload-monitoring-config
  namespace: openshift-user-workload-monitoring
data:
  config.yaml: |
    thanosRuler:
      topologySpreadConstraints:
      - maxSkew: 1 <1>
        topologyKey: monitoring <2>
        whenUnsatisfiable: ScheduleAnyway <3>
        labelSelector:
          matchLabels: <4>
            app.kubernetes.io/name: thanos-ruler
----
```

Reply:
Yes, thank you for the observation! I made it like this because the ROSA and OSD distros will only include the user defined examples, so that is why the inclusion is kind of duplicated. I was thinking about this and how to make it less of a case for the OCP. I have some ideas so I will try to redo it a little more and let you know!

Reply:
I mean for me is ok because they are two different things, but just to let you know that are the same just changing the namespace for the used-defined